File: binary-tree-right-side/binary-tree-right-side.py

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        self.sideView(root, 0, ans)
        return ans
    def sideView(self, root, level, ans):
        if not root: return;
        if len(ans) == level: ans.append(root.val)
        self.sideView(root.right, level+1, ans)
        self.sideView(root.left, level+1, ans)
        # binary tree right

File: binary-tree-right-side/binary-tree-right-side.java

class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> ans = new ArrayList<>();
        sideView(root, 0, ans);
        return ans;
    }
    void sideView(TreeNode root, int level, List<Integer> ans){
        if(root==null) return;
        if(ans.size() == level) ans.add(root.val);
        sideView(root.right, level+1, ans);
        sideView(root.left, level+1, ans);
    }
}

// binary tree java